File: test_analytics.py
#!/usr/bin/env python3
"""
Test script to verify SochDB analytics integration.

This sends several test events to PostHog to verify the integration works.
Events should appear in PostHog dashboard at: https://us.posthog.com
"""

import os
import time

# Ensure analytics is enabled
os.environ['SOCHDB_DISABLE_ANALYTICS'] = 'false'

from sochdb.analytics import (
    capture,
    capture_error,
    track_database_open,
    track_vector_search,
    track_batch_insert,
    is_analytics_disabled,
    _get_anonymous_id
)

def main():
    print("=" * 60)
    print("SochDB Analytics Test")
    print("=" * 60)
    print()
    
    # Check status
    print(f"Analytics disabled: {is_analytics_disabled()}")
    print(f"Anonymous ID: {_get_anonymous_id()}")
    print()
    
    print("Sending test events...")
    print()
    
    # Test 1: Basic event
    print("1. Basic event...")
    capture('analytics_test', {
        'test_number': 1,
        'test_name': 'basic_event',
        'timestamp': time.time()
    })
    print("   ✅ Sent")
    
    # Test 2: Database open
    print("2. Database open event...")
    track_database_open('./test_db', 'embedded')
    print("   ✅ Sent")
    
    # Test 3: Vector search
    print("3. Vector search event...")
    track_vector_search(dimension=1536, k=10, latency_ms=45.7)
    print("   ✅ Sent")
    
    # Test 4: Batch insert
    print("4. Batch insert event...")
    track_batch_insert(count=1000, dimension=768, latency_ms=123.4)
    print("   ✅ Sent")
    
    # Test 5: Error event
    print("5. Error event...")
    capture_error('test_error', 'This is a test error for analytics verification')
    print("   ✅ Sent")
    
    print()
    print("=" * 60)
    print("All test events sent successfully!")
    print()
    print("Check PostHog dashboard:")
    print("https://us.posthog.com/project/YOUR_PROJECT_ID/events")
    print()
    print("Expected events:")
    print("  - analytics_test")
    print("  - database_opened")
    print("  - vector_search")
    print("  - batch_insert")
    print("  - error")
    print("=" * 60)

if __name__ == '__main__':
    main()

File: test_analytics.js
#!/usr/bin/env node
/**
 * Test script to verify SochDB analytics integration (Node.js).
 * 
 * This sends several test events to PostHog to verify the integration works.
 * Events should appear in PostHog dashboard at: https://us.posthog.com
 */

// Ensure analytics is enabled
process.env.SOCHDB_DISABLE_ANALYTICS = 'false';

const {
  captureAnalytics,
  captureError,
  trackDatabaseOpen,
  trackVectorSearch,
  trackBatchInsert,
  isAnalyticsDisabled
} = require('./sochdb-js/dist/cjs/index.js');

async function main() {
  console.log('='.repeat(60));
  console.log('SochDB Analytics Test (Node.js)');
  console.log('='.repeat(60));
  console.log();
  
  // Check status
  console.log(`Analytics disabled: ${isAnalyticsDisabled()}`);
  console.log();
  
  console.log('Sending test events...');
  console.log();
  
  try {
    // Test 1: Basic event
    console.log('1. Basic event...');
    await captureAnalytics('analytics_test_js', {
      test_number: 1,
      test_name: 'basic_event_js',
      timestamp: Date.now()
    });
    console.log('   ✅ Sent');
    
    // Test 2: Database open
    console.log('2. Database open event...');
    await trackDatabaseOpen('./test_db', 'embedded');
    console.log('   ✅ Sent');
    
    // Test 3: Vector search
    console.log('3. Vector search event...');
    await trackVectorSearch(1536, 10, 45.7);
    console.log('   ✅ Sent');
    
    // Test 4: Batch insert
    console.log('4. Batch insert event...');
    await trackBatchInsert(1000, 768, 123.4);
    console.log('   ✅ Sent');
    
    // Test 5: Error event
    console.log('5. Error event...');
    await captureError('test_error_js', 'This is a test error for analytics verification');
    console.log('   ✅ Sent');
    
    console.log();
    console.log('='.repeat(60));
    console.log('All test events sent successfully!');
    console.log();
    console.log('Check PostHog dashboard:');
    console.log('https://us.posthog.com/project/YOUR_PROJECT_ID/events');
    console.log();
    console.log('Expected events:');
    console.log('  - analytics_test_js');
    console.log('  - database_opened');
    console.log('  - vector_search');
    console.log('  - batch_insert');
    console.log('  - error');
    console.log('='.repeat(60));
  } catch (err) {
    console.error('Error:', err);
    process.exit(1);
  }
}

main();
